### PR TITLE
feat: add toString to futures returned by operations

### PR DESCRIPTION
**Pull request by igorbernstein2**
Sometimes an operation can get stuck indefinitely. The underlying reasons can vary significantly:
- the underlying attempt rpc can get stuck due to a bug in grpc (ie https://github.com/grpc/grpc-java/pull/11026)
- the operation can get stuck in layers above gax: https://github.com/googleapis/java-bigtable/pull/1939
- or it can get stuck in gax itself (dont have a pointer handy)

Guava futures provide some observability for ListenableFutures, but in creating the custom ApiFutures in gax, we lose that functionality. This PR sprinkles a few to toString to allow callers to inspect the internal state of the operation. For example with these changes, the toString() of the future returned from bigtableDataClient.mutateRows() changes from

> TransformFuture@652ce654[status=PENDING, info=[inputFuture=[com.google.api.core.ApiFutureToListenableFuture@522ba524], function=[com.google.api.core.ApiFutures$ApiFunctionToGuavaFunction@29c5ee1d]]]

to
> ListenableFutureToApiFuture{delegate=TransformFuture@7ac9af2a[status=PENDING, info=[inputFuture=[ApiFutureToListenableFuture{apiFuture=CallbackChainRetryingFuture{super=com.google.api.gax.retrying.CallbackChainRetryingFuture@7bb004b8[status=PENDING], latestCompletedAttemptResult=null, attemptResult=null, attemptSettings=TimedAttemptSettings{globalSettings=RetrySettings{totalTimeout=PT10M, initialRetryDelay=PT0.01S, retryDelayMultiplier=2.0, maxRetryDelay=PT1M, maxAttempts=0, jittered=true, initialRpcTimeout=PT1M, rpcTimeoutMultiplier=1.0, maxRpcTimeout=PT1M}, retryDelay=PT0S, rpcTimeout=PT1M, randomizedRetryDelay=PT0S, attemptCount=0, overallAttemptCount=0, firstAttemptStartTimeNanos=635709620001791}}}], function=[com.google.api.core.ApiFutures$ApiFunctionToGuavaFunction@652ce654]]]}

This allows us to reason about whats stuck. I'm working another PR that will add a close(timeout) to the Batcher that will use this functionality to identify why batcher.close() timed out

Thank you for opening a Pull Request! Before submitting your PR, please read our [contributing guidelines](https://github.com/googleapis/gapic-generator-java/blob/main/CONTRIBUTING.md). 

There are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/gapic-generator-java/issues/new/choose) before writing your code! That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: api-common-java/src/test/java/com/google/api/core/ApiFutureToListenableFutureTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ */
+package com.google.api.core;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
```
Comment (blakeli0) [inline api-common-java/src/test/java/com/google/api/core/ApiFutureToListenableFutureTest.java:38]: The `api-common-java` module is migrated to Junit 5 now, can you please change this test to Junit 5 syntax?

Comment (igorbernstein2) [reply api-common-java/src/test/java/com/google/api/core/ApiFutureToListenableFutureTest.java]: done

File: api-common-java/src/test/java/com/google/api/core/ApiFutureToListenableFutureTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.core;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+public class ApiFutureToListenableFutureTest {
```
Comment (blakeli0) [inline api-common-java/src/test/java/com/google/api/core/ApiFutureToListenableFutureTest.java:36]: Junit 5 discourages the use of `public` in tests classes and methods, see [another test](https://github.com/googleapis/sdk-platform-java/blob/e5d23d4f3021c7957821b8bac10e43ea952b28d1/api-common-java/src/test/java/com/google/api/core/ListenableFutureToApiFutureTest.java) for example.

Comment (igorbernstein2) [reply api-common-java/src/test/java/com/google/api/core/ApiFutureToListenableFutureTest.java]: fixed!

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**